helm-eshell.el

```diff
@@ -71,6 +77,7 @@
    (nomark :initform t)
    (persistent-action :initform 'ignore)
    (nohighlight :initform t)
+   (fuzzy-match :initform (eval helm-eshell-fuzzy-match))
```

Comment:
I think it is better to run `:before`. Otherwise it breaks classical helm completion.

```
(defmethod helm--setup-source :before ((source helm-esh-source))
  (setf (slot-value source 'fuzzy-match) helm-eshell-fuzzy-match))
```